### ISSUE TITLE
Error in match.names(clabs, names(xi)) : names do not match previous names

### ISSUE THREAD

**Issue by LexRandea** [stefvanbuuren/mice#112]
Dear MICE enthusiasts,

I try to impute a real-life dataset with the mice 3.1.0 package in R 3.5 (with Rstudio)  but get the following cryptic error:  Error in match.names(clabs, names(xi)) : names do not match previous names (more details below).

Because I don't want to share my actual data set, I generated code (see below) which produces the same error for a randomly generated data set. In this example code, I suspect that the error occurs because of an insufficient number of samples to impute the variables leading to numerical issues. However, in my actual data set I suspect the problem to be more subtle, but still numerical in nature.

Any details about the cause of this error and how to avoid/solve it would be greatly appreciated. If this error turns out not to be MICE specific or is already solved, I would like to apologize in advance.

Kind regards,

Lex

**In short:** 
mice produces the following error about variable names "Error in match.names(clabs, names(xi)) : names do not match previous names" while I suspect the real issue is related to numerical problems with the imputation. If possible, I suggest catching the error in advance and produce a more informative error which aids the user in understanding the numerical issue. 

**Sample code:**
See attachment [example_R_code.txt](https://github.com/stefvanbuuren/mice/files/2138303/example_R_code.txt) (R code change extension)

**Error details:**
Error in match.names(clabs, names(xi)) : names do not match previous names
 1: source("~/Script_Simple_MICE.R", echo = TRUE)
 2: withVisible(eval(ei, envir))
 3: eval(ei, envir)
 4: eval(ei, envir)
 5: Script_Simple_MICE.R#45: mice(df, m = 1, maxit = 1, meth = alg, seed = 1, nnet.MaxNWts = 2000)
 6: sampler(data, m, where, imp, blocks, method, visitSequence, predictorMatrix, formulas, blots, post
 7: sampler.univ(data = data, r = r, where = where, type = type, formula = ff, method = theMethod, yna
 8: check.df(x, y, ry)
 9: updateLog(out = mess, frame = 4)
10: rbind(r, rec)
11: base::rbind(...)
12: rbind(deparse.level, ...)
13: match.names(clabs, names(xi))

**Comment by gerkovink** [comment]
Dear Lex, 

The `alg` object that you parse to argument `method` is a list. Argument `method` expects either a string or a vector of strings. If you replace the mice call with  the following line, it runs fine. 

```
res <- mice(df, m=1, maxit=1, meth=unlist(alg), seed=1, nnet.MaxNWts = 2000)
```

All the best, 

Gerko

**Comment by LexRandea** [comment]
Dear Gerko,

Thanks for your quick reply. I was really put on the wrong track by the fact that the function runs without errors with a list as `method` argument as long as there are no events added to the log. In the future, I try to read the documentation better. 

Thanks again and kind regards,

Lex

**Comment by gerkovink** [comment]
Dear Lex, 

Calling `method[index]` has no different behaviour if `method` is a list cf. your specs or a vector. However, at some point in the algorithm, the system is computationally singular. `mice` fixes this automatically and informs the users about the solution it has used by updating the `loggedEvents`. To do so, it uses `base::rbind()`, which breaks if `method` is not a vector. 

To at least make the behaviour consistent this is solved in #113 

All the best, 

Gerko
